diesel/src/types/mod.rs

```diff
@@ -274,10 +275,10 @@ pub type VarChar = Text;
 pub use pg::types::sql_types::*;
 
 pub trait HasSqlType<ST>: TypeMetadata {
-    fn metadata() -> Self::TypeMetadata;
+    fn metadata(lookup: &Self::MetadataLookup) -> Self::TypeMetadata;
```

Comment:
You're right, either one could work. I felt that this version was more flexible, and required less churn.